platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java

```diff
@@ -439,23 +468,66 @@ public void onFatalError() {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
     @Override
-    public AutoCloseableWrapper<SignedState> find(final long round, final Hash hash) {
-        return signedStateManager.find(round, hash);
+    public AutoCloseableWrapper<SignedState> find(final @NonNull Predicate<SignedState> criteria) {
+        return signedStateManager.find(criteria);
     }
 
     /**
      * This observer is called when the most recent signed state is requested to be dumped to disk.
      *
+     * @param round    the round that should be dumped if available. If this parameter is null or if the requested round
+     *                 is unavailable then the latest immutable round should be dumped.
      * @param reason   reason why the state is being dumped, e.g. "fatal" or "iss". Is used as a part of the file path
      *                 for the dumped state files, so this string should not contain any special characters or
      *                 whitespace.
      * @param blocking if this method should block until the operation has been completed
      */
     @Observer(StateDumpRequestedTrigger.class)
-    public void stateDumpRequestedObserver(final String reason, final Boolean blocking) {
+    public void stateDumpRequestedObserver(
+            @Nullable final Long round, @NonNull final String reason, @NonNull final Boolean blocking) {
```

Comment:
if `blocking` is defined as `NonNull` why not directly using the primitive datatype `boolean`?